src/core/net/srp_server.hpp

```diff
@@ -317,6 +333,15 @@ class Server : public InstanceLocator, private NonCopyable
          */
         TimeMilli GetKeyExpireTime(void) const;
 
+        /**
+         * This method returns the LEASE and KEY-LEASE information of a given service.
+         *
+         * @returns  An otSrpServerLeaseInfo of the service. It contains the LEASE time, KEY-LEASE time, remaining LEASE
+         * time and the remaining KEY-LEASE time.
+         *
+         */
+        otSrpServerLeaseInfo GetLeaseInfo(void) const;
```

Comment:
Same suggestion here. Instaead of returning the type, get a refernce to populate.

Also suggest defining a related C++ type here in `SrpServer` class itself and use it instead of the C definition. Something like:
```
typedef otSrpServerLeaseInfo  LeasseInfo;
```